src/_pytest/_code/code.py

```diff
@@ -469,12 +469,20 @@ def __init__(
         self._traceback = traceback
 
     @classmethod
-    def from_exc_info(
+    def from_exception(
         cls,
-        exc_info: Tuple[Type[E], E, TracebackType],
+        # Ignoring error: "Cannot use a covariant type variable as a parameter".
+        # This is OK to ignore because this class is (conceptually) readonly.
+        # See https://github.com/python/mypy/issues/7049.
+        exception: E,  # type: ignore[misc]
         exprinfo: Optional[str] = None,
     ) -> "ExceptionInfo[E]":
-        """Return an ExceptionInfo for an existing exc_info tuple.
+        """Return an ExceptionInfo for an existing exception.
+
+        The exception must have a non-``None`` ``__traceback__`` attribute,
```

Comment:
The only hitch here is this subtlety. An alternative is to just allow ExcInfo to have a missing traceback...

Reply:
Hmm I personally prefer to get a clear error at this point, rather than make `ExceptionInfo.__traceback__` be `Optional`.

Reply:
Yes, agreed.